src/typechecker.erl

```diff
@@ -782,6 +780,8 @@ normalize({op, _, _, _Arg} = Op, _Env) ->
     erl_eval:partial_eval(Op);
 normalize({op, _, _, _Arg1, _Arg2} = Op, _Env) ->
     erl_eval:partial_eval(Op);
+normalize({ann_type, Ann, [Var, Ty]}, Env) ->
+    {ann_type, Ann, [Var, normalize(typelib:remove_pos(Ty), Env)]};
```

Comment:
`remove_pos` is deep so I don't see why it's needed here. It's only done in normalize where we expand remote and user-defined types.

Reply:
Fixed :+1: